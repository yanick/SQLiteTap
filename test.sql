.load sql_tap.ext

select plan(4);

select ok( 1, "this passes" );
select ok( 0, "this fails" );

create table puppies (
    name text,
    cuteness number
);

insert into puppies values ( 'lassie', 5 );
insert into puppies values ( 'spot', 7 );

select ok( cuteness >= 5, name || ' is darn cute' ) from puppies;
