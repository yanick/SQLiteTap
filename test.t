use strict;
use warnings;

use Test::More tests => 4;

use DBI;

my $dbh = DBI->connect( 'dbi:SQLite:test.db' );

$dbh->sqlite_enable_load_extension(1);
$dbh->do("select load_extension('sql_tap.ext')");

sub db_ok($$$) {
    my ( $query, $result, $desc ) = @_;
    my $sth = $dbh->prepare( $query );
    $sth->execute;
    is $sth->fetchrow => $result, $desc;
}

db_ok 'select plan(4)', '1..4', 'plan';

$sth = $dbh->prepare( 'select ok( 1, "this passes" )' );
$sth->execute;
is $sth->fetchrow => 'ok 1 - this passes', "ok with desc";

$sth = $dbh->prepare( 'select ok( 0, "this fails" )' );
$sth->execute;
is $sth->fetchrow => 'not ok 2 - this fails', "not ok with desc";


