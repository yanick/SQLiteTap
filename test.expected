1..4
ok 1 - this passes
not ok 2 - this fails
ok 3 - lassie is darn cute
ok 4 - spot is darn cute
ok 5
not ok 6
